docker/gcp_pudl_etl.sh

```diff
@@ -100,60 +95,95 @@ function notify_slack() {
 }
 
 function update_nightly_branch() {
-    git config --unset http.https://github.com/.extraheader
-    git config user.email "pudl@catalyst.coop"
-    git config user.name "pudlbot"
-    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git"
-    echo "BOGUS: Updating nightly branch to point at $NIGHTLY_TAG."
-    git fetch origin nightly:nightly
-    git checkout nightly
-    git merge --ff-only "$NIGHTLY_TAG"
-    ETL_SUCCESS=${PIPESTATUS[0]}
-    git push -u origin
+    git config --unset http.https://github.com/.extraheader && \
+    git config user.email "pudl@catalyst.coop" && \
+    git config user.name "pudlbot" && \
+    git remote set-url origin "https://pudlbot:$PUDL_BOT_PAT@github.com/catalyst-cooperative/pudl.git" && \
+    echo "Updating nightly branch to point at $NIGHTLY_TAG." && \
+    git tag -d "$NIGHTLY_TAG" && \
+    git fetch --tags origin "$NIGHTLY_TAG" && \
+    git checkout "$NIGHTLY_TAG" && \
+    git tag && \
+    git fetch origin nightly:nightly && \
+    git checkout nightly && \
+    git merge --ff-only "$NIGHTLY_TAG^0" && \
+    git push -u origin nightly
+}
+
+function clean_up_outputs_for_distribution() {
+    # Compress the SQLite DBs for easier distribution
+    gzip --verbose "$PUDL_OUTPUT"/*.sqlite && \
+    # Remove redundant multi-file EPA CEMS outputs prior to distribution
+    rm -rf "$PUDL_OUTPUT/core_epacems__hourly_emissions/" && \
+    rm -f "$PUDL_OUTPUT/metadata.yml"
 }
 
-# # Run ETL. Copy outputs to GCS and shutdown VM if ETL succeeds or fails
+########################################################################################
+# MAIN SCRIPT
+########################################################################################
+# Initialize our success variables so they all definitely have a value to check
+ETL_SUCCESS=0
+SAVE_OUTPUTS_SUCCESS=0
+UPDATE_NIGHTLY_SUCCESS=0
+DATASETTE_SUCCESS=0
+CLEAN_UP_OUTPUTS_SUCCESS=0
+DISTRIBUTION_BUCKET_SUCCESS=0
+ZENODO_SUCCESS=0
```

Comment:
Using separate variables and checking that they are all 0 before declaring success at the end. Could probably do some better communication of why things failed if they fail.

Reply:
WOW this is a great idea and I can't believe I hadn't thought of this. Will definitely remember that there's a whole slew of steps & separate states when we port to Python!

Reply:
I suppose we could have different numbers for "not started, attempted, succeeded, failed". But we can do without that for now.